modules/mysql.lua

```diff
@@ -6,35 +6,44 @@
 local l = require "lpeg"
 l.locale(l)
 local tonumber = tonumber
+local ip = require "ip_address"
 
 local M = {}
 setfenv(1, M) -- Remove external access to contain everything in the module
 
+local pct_encoded   = l.P"%" * l.xdigit * l.xdigit
+local unreserved    = l.alnum + l.S"-._~"
+local sub_delims    = l.S"!$&'()*+,;="
+
 local space         = l.space^1
 local sep           = l.P"\n"
 local sql_end       = l.P";" * (l.P"\n" + -1)
 local line          = (l.P(1) - sep)^0 * sep
-local float         = l.digit^1 * "." * l.digit^1
+local float         = l.digit^1 * "." * l.digit^1 / tonumber
+local integer       = l.P"-"^-1 * l.digit^1 / tonumber
+local host          = l.Ct(l.Cg(ip.v4, "value") * l.Cg(l.Cc"ipv4", "representation"))
+                    + l.Ct(l.Cg(ip.v6, "value") * l.Cg(l.Cc"ipv6", "representation"))
+                    + l.Ct(l.Cg((unreserved + pct_encoded + sub_delims)^1, "value") * l.Cg(l.Cc"hostname", "representation"))
 
 local time          = l.P"# Time: " * line
 
 local user_legal    = (1 - l.S"[]")
 local host_legal    = (l.alnum + l.S".-")
-local user_name     = user_legal^1 * "[" * l.Cg(user_legal^1, "Username") * "]"
-local host_name     = host_legal^0 * l.space^0 * "[" * l.Cg((l.P(1) - "]")^1, "Hostname") * "]"
+local user_name     = user_legal^0 * "[" * l.Cg(user_legal^0, "Username") * "]"
+local host_name     = l.Cg(host, "Client_hostname")^0 * l.space^0 * "[" * l.Cg(host, "Client_address")^0 * "]"
 local user          = l.P"# User@Host: " * user_name * space * "@" * space * host_name * sep
 
-local query_time    = l.P"# Query_time: " * l.Cg(l.Ct(l.Cg(float / tonumber, "value") * l.Cg(l.Cc"s", "representation")), "Query_time")
-local lock_time     = l.P"Lock_time: " * l.Cg(l.Ct(l.Cg(float / tonumber, "value") * l.Cg(l.Cc"s", "representation")), "Lock_time")
-local rows_sent     = l.P"Rows_sent: " * l.Cg(l.digit^1 / tonumber, "Rows_sent")
-local rows_examined = l.P"Rows_examined: " * l.Cg(l.digit^1 / tonumber, "Rows_examined")
+local query_time    = l.P"# Query_time: " * l.Cg(l.Ct(l.Cg(float, "value") * l.Cg(l.Cc"s", "representation")), "Query_time")
+local lock_time     = l.P"Lock_time: " * l.Cg(l.Ct(l.Cg(float, "value") * l.Cg(l.Cc"s", "representation")), "Lock_time")
+local rows_sent     = l.P"Rows_sent: " * l.Cg(integer, "Rows_sent")
+local rows_examined = l.P"Rows_examined: " * l.Cg(integer, "Rows_examined")
 local query         = query_time * space * lock_time * space * rows_sent * space * rows_examined * sep
 
 local use_db        = l.P"use " * line
 
-local last_insert   = l.P"last_insert_id=" * l.digit^1 * ","
-local insert        = l.P"insert_id=" * l.digit^1 * ","
-local timestamp     = l.P"timestamp=" * l.Cg((l.digit^1 / "%0000000000"), "Timestamp")
+local last_insert   = l.P"last_insert_id=" * l.Cg(integer, "Last_insert") * ","
+local insert        = l.P"insert_id=" * l.Cg(integer, "Insert_id") * ","
+local timestamp     = l.P"timestamp=" * l.Cg(integer / "%0000000000", "Timestamp")
```

Comment:
Just saw that I missed this; changing it back to the original timestamp pattern.